controllers/datamovement_controller_test.go

```diff
@@ -91,7 +178,404 @@ var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
 		})
 	})
 
+	Context("when the dm configmap has specified a dmCommand", func() {
+		BeforeEach(func() {
+			cmData[configMapKeyCmd] = "ls -l"
+		})
+		It("should use that command instead of the default mpirun", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal("/bin/" + cmData[configMapKeyCmd]))
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmNumProcesses and the dm namespace matches NNF_NODE_NAME", func() {
+		BeforeEach(func() {
+			cmData[configMapKeyCmd] = ""
+			cmData[configMapKeyNumProcesses] = "17"
+			os.Setenv("NNF_NODE_NAME", "default") // normally this isn't default, but we just want it to match
+		})
+
+		It("should use that number for the -np flag", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 17 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+
+		AfterEach(func() {
+			os.Unsetenv("NNF_NODE_NAME")
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmNumProcesses and the dm namespace does not match NNF_NODE_NAME", func() {
+		BeforeEach(func() {
+			cmData[configMapKeyCmd] = ""
+			cmData[configMapKeyNumProcesses] = "17"
+		})
+
+		It("should use len(hosts) (1) for the -np flag", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 1 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+	})
+
+	Context("when the dm config map has specified a invalid dmNumProcesses", func() {
+		BeforeEach(func() {
+			cmData[configMapKeyCmd] = ""
+			cmData[configMapKeyNumProcesses] = "aa"
+		})
+
+		It("should use the default number for -np flag", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 1 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmProgressInterval", func() {
```

Comment:
These three tests might be better represented by the [Ginkgo Table Spec](https://onsi.github.io/ginkgo/#table-specs) since the main It() block is the same across each, but the expectations are different.

Reply:
dcpProgressInterval tests too

Reply:
That looks nice and convenient. I'll give it a go.

Reply:
On second thought, I will do this in a subsequent PR.